### PR TITLE
TASK-51 Completed: Creation of a new page called leaderboard

### PR DESCRIPTION
**Pull request by Asafyaco**
Created a new page called **leader-board.page** - display all the top 10 users with the most score on the game

* Created a **new route of the LeaderBoardPage** on App.jsx
* Updated the translation both on English & Hebrew - on navBar and the title of the page
* Updated the translation.service
* Added to the navBarComponent in the floating button the page - **LeaderBoard** 
* Integrate with an API on the backend
* Added on states on constants.ts - **/leaderBoard**

Example of the page -
image

### CODE REVIEW

**Comment by Asafyaco** [comment]
* Added an emoji next to the title of the page on **leaderBoardPage**
* Added media query to navBar component that **wrap all the links** and don't hide them depending on the device
* Updated dependencies on the **leaderBoardPage.scss** 

Example of warped navBar on different devices:
image

Example of floating button on navbar:
image

File: src/pages/leader-board/leader-board.scss [hunk 1]
```diff
@@ -0,0 +1,61 @@
+.qcg-leader-board-page {
+    font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
+    text-transform: uppercase;
+    overflow-y: scroll;
+    line-height: normal;
+    height: 80vh;
```
Comment (FadiFarah) [inline src/pages/leader-board/leader-board.scss]: change the height to height: calc(100vh - 10em); as we did on the other pages for the scroll to work properly.

File: src/pages/leader-board/leader-board.page.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    useEffect(() =>{
+      getAllLeaderBoard();
+    },[null]);
+
+  return (
+  <div className="qcg-leader-board-page qcg-flex qcg-flex-center">
+    <div className="wrapper">
+      <div className="board-wrapper qcg-flex qcg-flex-column qcg-flex-align-center">
+        <h1>{translationService.translate.leaderBoardPage.title}</h1>
+        {leaderboardUsers.length <= 10 ? (leaderboardUsers.map((top10, index)=>
```
Comment (FadiFarah) [inline src/pages/leader-board/leader-board.page.jsx]: change the naming of top10 to user.

Comment (FadiFarah) [inline src/pages/leader-board/leader-board.page.jsx]: We don't need an if statement. We always get from the database maximum 10 users, so need to check. Remove it.

File: src/pages/leader-board/leader-board.page.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        {return <div className="circle-around qcg-flex qcg-flex-row qcg-flex-align-center" key={index}>
+          <div className="id">
+          {index+1}
+          </div>
+          <img src={top10.picture} alt="userProfilePicture" />
+          <div className="username"> {top10.firstName} </div>
+          <div className="points"> {top10.score}</div>
+          </div>})):(
+             <div></div>
+           )}
```
Comment (FadiFarah) [inline src/pages/leader-board/leader-board.page.jsx]: Remove the unneeded empty div with the above multiple opening and closing brackets.

File: src/assets/translations/english.json [hunk 4]
```diff
@@ -8,6 +8,7 @@
       "about": "about",
       "rules": "rules",
       "profile": "profile",
+      "leaderBoard":"leader board",
```
Comment (FadiFarah) [inline src/assets/translations/english.json]: remove the space between the two words only for English.

File: src/pages/leader-board/leader-board.scss [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            margin-right: 1rem;
+            border-radius: 50%;
+          } 
+        }
+      }
+    }
+
+    @media only screen and (max-width: 940px) and (max-height: 660px) {
+      &.qcg-leader-board-page {
+        height: 55vh;
```
Comment (FadiFarah) [inline src/pages/leader-board/leader-board.scss]: No need for height here, remove it.

**Review by FadiFarah — CHANGES_REQUESTED** [review]
One additional thing to change... the leaderboard should only be visible to authenticated users only, just as the profile.

File: src/pages/leader-board/leader-board.scss (hunk 1 above)
Comment (Asafyaco) [reply src/pages/leader-board/leader-board.scss]: 👍

File: src/pages/leader-board/leader-board.page.jsx (hunk 2 above)
Comment (Asafyaco) [reply src/pages/leader-board/leader-board.page.jsx]: Ohh ok, didn't remember that. Has been removed

**Comment by Asafyaco** [comment]
**Commit updated by comments**

* Changed height on leader-boards.scss
* Updated name on map function on leader-board.page
* Removed an if statement & Removed empty div
* Removed height on media query 
* Updated that the leader board page will shown only to the authenticated users
* Updated translation of first letter of the links on floating button on navBarComponent to a capital letter

**Review by FadiFarah — APPROVED** [review]

**Pull request merged**